src/libstd/time/duration.rs

```diff
@@ -139,6 +139,27 @@ impl Duration {
         Duration { secs: secs, nanos: nanos }
     }
 
+    /// Creates a new `Duration` from the specified number of nanoseconds.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(duration_extras)]
+    /// use std::time::Duration;
+    ///
+    /// let duration = Duration::from_nanos(1_000_000_123);
+    ///
+    /// assert_eq!(1, duration.as_secs());
+    /// assert_eq!(123, duration.subsec_nanos());
+    /// ```
+    #[unstable(feature = "duration_extras", issue = "46507")]
+    #[inline]
+    pub fn from_nanos(nanos: u64) -> Duration {
```

Comment:
This feels like it should be `from_nanosecs` or something like that by the convention in other methods.

Reply:
The existing constructors are `new`, `from_secs`, `from_millis`, and `from_micros`, so it seems like `from_nanos` would match conventions?

Reply:
I guess we've been somewhat inconsistent with abbreviating seconds already, so this seems fine.

Reply:
I personally like the use of the pluralised prefices, considering how simply using `ns`, `us`, and `ms` would be very easy to make typos. I don't personally like using `sec` as an abbreviation for `second` considering how `sec` is secant and `s` is second, but that's just because I'm a maths person and I appreciate clarity in equations. We shouldn't really change the naming scheme now that this has been stabilised for a long time.